scrypto/src/types/decimal.rs

```diff
@@ -99,6 +99,46 @@ from_int!(i64);
 from_int!(i128);
 from_int!(isize);
 
+impl From<&str> for Decimal {
+    fn from(val: &str) -> Self {
+        Self::from_str(&val).unwrap()
+    }
+}
+
+impl From<String> for Decimal {
+    fn from(val: String) -> Self {
+        Self::from_str(&val).unwrap()
+    }
+}
+
+impl From<bool> for Decimal {
+    fn from(val: bool) -> Self {
+        if val {
+            Self::from(1)
+        } else {
+            Self::from(0)
+        } 
+    }
+}
+
+#[macro_export]
+macro_rules! dec {
+    
+    ($x:literal) => {
+       Decimal::from($x) 
+    };
+    
+    ($int:literal, $exponent:literal) => {
+        if ($exponent) < 0 {
+            Decimal(($int) * PRECISION)
+                .div(10i128.pow((-1i128 * ($exponent)) as u32))
+        } else {
+            Decimal(($int) * PRECISION)
+                .mul(10i128.pow((1i128 * ($exponent)) as u32))
```

Comment:
`as u32` will silence underflow/overflow instead of panic. I'd suggest we force a check before doing the calculation and add a test case to cover this edge case.

Reply:
@iamyulong I have updated the code according to your remarks. Please review again.